libs/wallet/src/web3-react/updater.ts

```diff
@@ -65,40 +70,80 @@ function _useSafeInfo(walletInfo: WalletInfo): GnosisSafeInfo | undefined {
   const { provider } = useWeb3React()
   const { account, chainId } = walletInfo
   const [safeInfo, setSafeInfo] = useState<GnosisSafeInfo>()
-  const { isReadOnly } = useSafeAppsSdkInfo() || {}
+  const safeAppsSdk = useSafeAppsSdk()
 
   useEffect(() => {
-    const update = () => {
+    const updateSafeInfo = async () => {
       if (chainId && account && provider) {
-        getSafeInfo(chainId, account, provider)
-          .then((_safeInfo) =>
-            setSafeInfo({
+        if (safeAppsSdk) {
+          const appsSdkSafeInfo = await safeAppsSdk.safe.getInfo()
+          setSafeInfo((prevSafeInfo) => {
+            const { isReadOnly, nonce } = appsSdkSafeInfo
```

Comment:
As much as I prefer async/await instead of promise chaining, this is:
1. Not in the same style as the way it's fetched in the `else` clause
2. Not handling the failure case AFAICT

Reply:
Yes, you are right. I just copied the old code that was using promises to the else block and added my code with await. I've changed this now, so both blocks are using async await syntax.